### PR TITLE
[freetype] Update to 2.10.4

### PR DESCRIPTION
**Pull request by kevinlul**
**Describe the pull request**

- What does your PR fix?

Regular port update https://sourceforge.net/projects/freetype/files/freetype2/2.10.4/ with major security vulnerability fix, plus the upstream vtk patch for compatibility with this revision from https://discourse.vtk.org/t/building-opencascade-7-5-0-with-vtk-9-0-1

Closes #14008. I made a new PR due to the changes on `master` and branched off the last commit that affected freetype and vtk since I could not build anything on HEAD #15332 but I could on 191e01f 

- Which triplets are supported/not supported? Have you updated the CI baseline?
It should work everywhere but since vtk takes really long to build, I've only tested against x86-windows.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Yes

### CODE REVIEW

File: ports/vtk/fix-freetype.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+diff --git a/Rendering/FreeTypeFontConfig/vtkFontConfigFreeTypeTools.cxx b/Rendering/FreeTypeFontConfig/vtkFontConfigFreeTypeTools.cxx
+index 5c1908cf..41ed97e7 100644
+--- a/Rendering/FreeTypeFontConfig/vtkFontConfigFreeTypeTools.cxx
++++ b/Rendering/FreeTypeFontConfig/vtkFontConfigFreeTypeTools.cxx
+@@ -29,7 +29,7 @@ vtkStandardNewMacro(vtkFontConfigFreeTypeTools);
+ namespace
+ {
+ // The FreeType face requester callback:
+-FT_CALLBACK_DEF(FT_Error)
++static FT_Error
```
Comment (JackBoosY) [inline ports/vtk/fix-freetype.patch:23]: What is wrong here? Does the upstream accept this changes and know this bug?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/vtk/fix-freetype.patch (hunk 1 above)
Comment (kevinlul) [reply ports/vtk/fix-freetype.patch:23]: See the linked thread for the details. The short version that I understand is that `FT_CALLBACK_DEF` is actually an internal freetype macro that got removed in 2.10.4, and the patches I incorporate and reference replace the usage of this macro with what it actually expands to.

**Review by JackBoosY — APPROVED** [review]
Did you test the usage?

**Comment by kevinlul** [comment]
Behaviour of freetype is fine for me since that was working in the old PR but I cannot really say anything meaningful about vtk since I didn't expect this problem with a seemingly unrelated port.

**Comment by vicroms** [comment]
Thanks for the PR!

**Pull request merged**